File: translation_output/no_test/gpt-4/codenet/java/python/p03632.py
Here is the translated code in Python:

```python
import sys

for line in sys.stdin:
    a, b, c, d = map(int, line.split())
    start = max(a, c)
    end = min(b, d)
    print(max(0, end - start))
```

This Python code reads lines from standard input (just like `System.in` in Java), splits each line into integers, and then calculates the overlapping length between interval [a, b] and interval [c, d]. The result is printed out, which is either the length of the overlap or 0 if there's no overlap.

File: translation_output/no_test/gpt-4/codenet/python/java/p03632.java
Here's the Python code translated into Java:

```java
import java.util.Scanner;

public class OverlappingSegments {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read integers A, B, C, and D as input
        int A = scanner.nextInt();
        int B = scanner.nextInt();
        int C = scanner.nextInt();
        int D = scanner.nextInt();
        
        // Calculate the answer
        int ans = Math.min(B, D) - Math.max(A, C);
        
        // Print the result based on the condition
        if (ans > 0) {
            System.out.println(ans);
        } else {
            System.out.println(0);
        }
        
        // Close the scanner
        scanner.close();
    }
}
```

Make sure to run this Java code within the context of a class, as Java is a class-based language and code must be executed within the context of a class and method.
